rpm/files.go

```diff
@@ -0,0 +1,103 @@
+package rpm
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"sync"
+
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+)
+
+// filesCache is used for concurrent access to the map containing layer.Hash -> map RPM files.
+// The value is a map to allow for quick member checking.
+type filesCache struct {
+	c  map[string]map[string]struct{}
+	mu *sync.Mutex
+}
+
+var fc *filesCache
+
+func init() {
+	fc = &filesCache{
+		c:  map[string]map[string]struct{}{},
+		mu: &sync.Mutex{},
+	}
+}
+
+// gc deletes the layer's entry from the map if the ctx is done, this ties the lifecycle of
+// the cached information to the request lifecycle to avoid excessive memory consumption.
+func (fc *filesCache) gc(ctx context.Context, key string) {
```

Comment:
Is it assumed each language scanner's context is the same? If not, is it possible we cache a layer, delete it, then have to re-read it because it was just deleted?

Reply:
https://github.com/quay/claircore/blob/main/indexer/layerscanner.go#L234 they all start off the same, I guess there is a chance they're changed before the function is called.